pkg/e2ecluster/e2ehelpers/curl.go

```diff
@@ -0,0 +1,107 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package e2ehelpers
+
+import (
+	"fmt"
+	"strconv"
+	"time"
+)
+
+const (
+	defaultConnectTimeout = 5 * time.Second
+	defaultMaxTime        = 20 * time.Second
+)
+
+type CurlResultFormat string
+
+const (
+	CurlResultFormatStats    CurlResultFormat = `time-> DNS: '%{time_namelookup}(%{remote_ip})', Connect: '%{time_connect}', Transfer '%{time_starttransfer}', total '%{time_total}'`
+	CurlResultFormatHTTPCode CurlResultFormat = "%%{http_code}"
+)
+
+type opts struct {
```

Comment:
nit: Rename this to `curlOpts`, given that the e2epackage will get other helpers too.